test/regridder_tests.jl

```diff
@@ -314,3 +315,67 @@ for FT in (Float32, Float64)
         end
     end
 end
+# test dataset truncation 
+@testset "test dataset truncation" begin
+    # Get the original dataset set up 
+    include(joinpath(pkgdir(ClimaCoupler), "artifacts", "artifact_funcs.jl"))
+    sst_data_all = joinpath(sst_dataset_path(), "sst.nc")
+    ds = NCDataset(sst_data_all, "r")
+    dates = ds["time"][:]
+    first_date = dates[1]
+    last_date = last(dates)
+
+    # set up comms_ctx 
+    device = ClimaComms.device()
+    comms_ctx = ClimaComms.context(device)
+    ClimaComms.init(comms_ctx)
+
+    # make path for truncated datasets 
+    COUPLER_OUTPUT_DIR = joinpath("experiments", "AMIP", "output", "tests")
+    mkpath(COUPLER_OUTPUT_DIR)
+
+    REGRID_DIR = joinpath(COUPLER_OUTPUT_DIR, "regrid_tmp", "")
+    mkpath(REGRID_DIR)
+
+    # values for the truncations 
+    time_start = 0.0
+    time_end = 1.728e6
+    date0test = ["18690101", "18700101", "19790228", "20220301", "20230101"]
+    for date in date0test
+        date0 = DateTime(date, dateformat"yyyymmdd")
+        sst_data = Regridder.truncate_dataset(sst_data_all, "test", REGRID_DIR, date0, time_start, time_end, comms_ctx)
+        ds_truncated = NCDataset(sst_data, "r")
+        new_dates = ds_truncated["time"][:]
+
```

Comment:
This is a great test for checking the dates in the truncated dataset, but it would still be good to add a test that checks that the variable in the truncated dataset is aligned with the new truncated dates. If we split `truncate_dataset` as suggested above, we can add something like:

```
all_data = ds["SST"][:, :, :]
new_data = ds_truncated["SST"][:, :, :]
first_index = find_bounding_dates(...)
@test new_data[:,:,1] ≈ all_data data[:,:,first_index]  
```